docs/vsphere_ci/windows-server-2022.pkr.hcl

```diff
@@ -0,0 +1,150 @@
+  default = "Administrator"
+}
+
+variable "vm-template-folder" {
+  type    = string
+  default = "windows-golden-images"
+}
+
+variable "vm-template-name" {
+  type    = string
+  default = "windows-server-2022-template-test"
+}
+
+variable "vmtools-iso-path" {
+  type    = string
+  default = "[WorkloadDatastore] windows-iso-images/vmtools-v11360-windows.iso"
+}
+
+variable "vsphere-cluster" {
+  type    = string
+  default = "Cluster-1"
+}
+
+variable "vsphere-datacenter" {
+  type    = string
+  default = "SDDC-Datacenter"
+}
+
+variable "vsphere-datastore" {
+  type    = string
+  default = "WorkloadDatastore"
+}
+
+variable "vsphere-password" {
+  type      = string
+  default   = "vcenter_password"
+  sensitive = true
+}
+
+variable "vsphere-server" {
+  type    = string
+  default = "vcenter.example.com"
+}
+
+variable "vsphere-user" {
+  type    = string
+  default = "vcenter_user"
+}
+
+source "vsphere-iso" "windows-server-2022" {
+  CPUs                 = "4"
+  RAM                  = "16384"
+  RAM_reserve_all      = true
+  cluster              = "${var.vsphere-cluster}"
+  communicator         = "ssh"
+  convert_to_template  = "true"
+  datacenter           = "${var.vsphere-datacenter}"
+  datastore            = "${var.vsphere-datastore}"
+  disk_controller_type = ["lsilogic-sas"]
+  firmware             = "bios"
+  floppy_files         = ["answer-files/autounattend.xml", "answer-files/unattend.xml", "scripts/authorized_keys", "scripts/install-vm-tools.cmd", "scripts/configure-vm-tools.ps1", "scripts/install-openssh.ps1"]
+  folder               = "${var.vm-template-folder}"
+  guest_os_type        = "windows9Server64Guest"
+  insecure_connection  = "true"
+  iso_paths            = ["${var.os-iso-path}", "${var.vmtools-iso-path}"]
+  network_adapters {
+    network      = "dev-segment"
+    network_card = "vmxnet3"
+  }
+  password         = "${var.vsphere-password}"
+  shutdown_command = "c:\\windows\\system32\\sysprep\\sysprep.exe /generalize /oobe /shutdown /unattend:a:\\unattend.xml"
+  ssh_password     = "${var.vm-elevated-password}"
+  ssh_timeout      = "10m"
+  ssh_username     = "${var.vm-elevated-user}"
+  storage {
+    disk_size             = "61440"
+    disk_thin_provisioned = true
+  }
+  username       = "${var.vsphere-user}"
+  vcenter_server = "${var.vsphere-server}"
+  vm_name        = "${var.vm-template-name}"
+  vm_version     = 15
+}
+
+build {
+  sources = ["source.vsphere-iso.windows-server-2022"]
+
+  provisioner "file" {
+    destination = "C:/"
+    source      = "scripts/rename-computer.ps1"
+  }
+
+  provisioner "powershell" {
+    elevated_password = "${var.vm-elevated-password}"
+    elevated_user     = "${var.vm-elevated-user}"
+    script            = "scripts/install-firewall-rules.ps1"
+  }
+
+  provisioner "powershell" {
+    elevated_password = "${var.vm-elevated-password}"
+    elevated_user     = "${var.vm-elevated-user}"
+    script            = "scripts/install-docker.ps1"
+  }
```

Comment:
It looks like the order of the provisioner steps changed in the translation to HCL. In the HCL code, Docker is installed before any run of `install-updates.ps1`. But in build.json, Docker was installed after the running `install-updates.ps1` twice.

Was the golden image built using the HCL config? Just want to confirm in case we have to rebuild the image.

Reply:
> But in build.json, Docker was installed after the running install-updates.ps1 twice

Good catch. I can fix that just to make things consistent.

>Was the golden image built using the HCL config? Just want to confirm in case we have to rebuild the image.

Yes the final image was built using the HCL config.

Reply:
@saifshaikh48 fixed. To be clear I did most of my testing with the HCL config given that was the way forward. I only made the `build.json` changes to make it easier to review.